src/Elastic.Apm.MongoDb/LICENSE.txt

```diff
@@ -0,0 +1,2 @@
+The code in this directory is based on the elastic-apm-mongo project by Vadim Hatsura (@vhatsura),
+licensed under the Apache 2.0 License. https://github.com/vhatsura/elastic-apm-mongo
```

Comment:
Might be good to use the file name LICENSE, for consistency with the existing LICENSE in solution root.

I think this content should be concatenated with the contents of `$(SolutionRoot)/LICENSE` and included in the nuget package. An MSBuild target might be a good approach to handle this.

Then, src/Directory.Build.Props needs to be changed to **not** include `$(SolutionRoot)/LICENSE` in the nuget package when a LICENSE file exists in the current project root.

Reply:
Regarding file name: 👍 done.

Regarding the other part: I'm a bit confused here. We have our NOTICE file which says what 3rd party things we use and that's already included in the package. This file talks about `this directory` (which in the context of a NuGet package doesn't have a meaning), so the text sounds like this applies to the source code and not to the binary we generate.

I don't see why we need to have both NOTICE and LICENSE with basically telling the same, and furthermore if we need both then I think we need to adapt the text here.

My understanding was that we keep the LICENSE file here in the folder to communicate that this code is based on vhatsura/elastic-apm-mongo and we added the NOTICE flag to communicate that our binaries are inherited from code from vhatsura/elastic-apm-mongo.

@gtback @russcam what am I missing here? I'm really not a license person, so maybe I just misunderstand something.

Reply:
Yep, I meant to reply to this yesterday, but @russcam got it right. Per our policies,

> All product builds ... must export the full original license and copyright text of any dependencies where required by the dependency’s license in the NOTICE.txt file

Broadly speaking, `LICENSE` contains copyright and license information for the code in the repo (including code we've "vendored" into the repo), and `NOTICE` contains information about sources built into a distribution. This is a bit of a special case, in that the code **is** in the repo (so is included in `LICENSE`) **and** is part of the distribution, so we include the same information in both. Note that the ASF guidelines do not require (and in fact recommend against) putting the text of dependencies' licenses in NOTICE, but our practice is to do just that. Since we include the full copyright and license text in LICENSE as well, it ends up looking the same. 

---

> This file talks about `this directory` (which in the context of a NuGet package doesn't really have a meaning), so the text sounds like this applies to the source code and not to the binary we generate.

Good catch! In the case of binary distributions (NuGet packages), it probably makes more sense to say "The code for `Elastic.Apm.MongoDB` and `Elastic.Apm.MongoDb.Tests`..."

Reply:
Thanks for the input on this.

So, I implemented @russcam's suggestion with concatenating `LICENSE` files with an msbuild target. [Here](https://github.com/elastic/apm-agent-dotnet/pull/1215/commits/d4e5d6368289faa2af360b7c8926714b27e9268b) is the commit.

In summary:

- We have the main `LICENSE` file which is untouched and shipped with all NuGet packages (except next point)
- If there is a `LICENSE_LOCAL` file. (happy to rename if someone comes up with something better) in a specific project, then the main `LICENSE` file won't be added to the project and msbuild target will concatenate the main `LICENSE` file with the `LICENSE_LOCAL` and create the `LICENSE` file for the given project, which will be included in the generated package.
- Originally I wanted to only have a `LICENSE` file in the specific folders as well, but this would be harder to concatenate, because 1) we'd need to know if it was already concatenated 2) we'd need to detect if the main `LICENSE` file changed 3) we'd need to know if the local `LICENSE` file changed.

Reply:
This looks good; wonder if we can introduce a newline between each concatenated section, as I think it'd combine better? The license separator could also be part of the content between concatenated sections. Not a big deal though, and could be addressed later.

I was thinking that we could have a `LICENSE` file in the project and concatenate with the repo license into another file e.g. `COMBINED_LICENSE`, then pack this into the nuget package as `LICENSE`, but it seems that nuget's behaviour is buggy with extensionless files and renaming as part of packing (https://github.com/NuGet/Home/issues/6351#issuecomment-478903940), so `LICENSE_LOCAL` is pragmatic.

Reply:
> wonder if we can introduce a newline between each concatenated section, as I think it'd combine better?

Good idea, I implemented this, now we have a newline between the sections.